Todo/AppDelegate.swift

```diff
@@ -0,0 +1,420 @@
+                onCancel: { () -> Void in
+                    // Force delete the database:
+                    let dir = NSURL(fileURLWithPath: CBLManager.sharedInstance().directory)
+                    let dbFile = dir.URLByAppendingPathComponent("\(username).cblite2")
+                    do {
+                        try NSFileManager.defaultManager().removeItemAtURL(dbFile)
+                    } catch let err as NSError {
+                        NSLog("Error when deleting the database file: %@", err)
+                    }
+                    // login:
+                    self.processLogin(controller, withUsername: username,
+                        withPassword: password)
+                }
+            )
+    }
+
+    // MARK: - Replication
+
+    func startReplication(withUsername username:String!, andPassword password:String? = "") {
+        if !kSyncEnabled {
+            return
+        }
+
+        var authenticator: CBLAuthenticatorProtocol?
+        var headers: [String: String]?
+        if kLoginFlowEnabled {
+            authenticator = CBLAuthenticator.basicAuthenticatorWithName(username, password: password!)
+
+            // Workaround #1124:
+            // https://github.com/couchbase/couchbase-lite-ios/issues/1124
+            let cred = NSString(format: "%@:%@", username, password!)
+            let credData = cred.dataUsingEncoding(NSUTF8StringEncoding)!
+            let credBase64 = credData.base64EncodedStringWithOptions(.Encoding64CharacterLineLength)
+            headers = ["Authorization": "Basic \(credBase64)"]
+        }
+        
+        syncError = nil
+        
+        pusher = database.createPushReplication(kSyncGatewayUrl)
+        pusher.continuous = true
+        pusher.authenticator = authenticator
+        pusher.headers = headers
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "replicationProgress:",
+            name: kCBLReplicationChangeNotification, object: pusher)
+
+        puller = database.createPullReplication(kSyncGatewayUrl)
+        puller.continuous = true
+        puller.authenticator = authenticator
+        puller.headers = headers
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "replicationProgress:",
+            name: kCBLReplicationChangeNotification, object: puller)
+
+        pusher.start()
+        puller.start()
+    }
+    
+    func stopReplication() {
+        if !kSyncEnabled {
+            return
+        }
+
+        pusher.stop()
+        NSNotificationCenter.defaultCenter().removeObserver(
+            self, name: kCBLReplicationChangeNotification, object: pusher)
+        
+        puller.stop()
+        NSNotificationCenter.defaultCenter().removeObserver(
+            self, name: kCBLReplicationChangeNotification, object: puller)
+    }
+    
+    func replicationProgress(notification: NSNotification) {
+        if (pusher.status == .Active || puller.status == .Active) {
```

Comment:
Just assign the boolean:

`UIApplication.sharedApplication().networkActivityIndicatorVisible = (pusher.status == .Active || puller.status == .Active)`